proposals/p2006.md

```diff
@@ -236,13 +255,44 @@ _thread-safe_ interface subset of an otherwise _thread-compatible_ type.
 > Details:
 > [`const`-qualified types](/docs/design/values.md#const-qualified-types)
 
-### Interop with `const &`
+### Interop with `const &` and `const` methods
 
-> TODO
+Carbon makes value expressions interoperable with `const &` function parameters
+and `const` methods in C++. This is tricky because those constructs in C++ _do_
+allow the address to be taken. As a consequence, Carbon will synthesize an
+object and address for value expressions when necessary for interop. This
+mechanism may also be used as an escape hatch for value expressions, but that is
+[currently not proposed](#value-expression-escape-hatches).
+
+This interop is based on the fact that these constructs in C++ can bind to
+`const`-declared objects which strongly pressure code to be compatible even with
+Carbon's more restrictive value expression semantics.
```

Comment:
I feel like I don't fully understand this sentence. Could you explain the steps in the reasoning a bit more explicitly for me?